### PR TITLE
bug: non-compliance messages only work for specific policy definition modes

### PR DESCRIPTION
**Pull request by jaredfholgate**
This adds a check for the policy definition mode to ensure the policy assignment can support non-compliance messages.

### CODE REVIEW

File: internal/provider/archetype_data_source.go [hunk 1]
```diff
@@ -689,23 +689,38 @@ func policyAssignmentType2ArmPolicyAssignment(pamap map[string]PolicyAssignmentT
...
 		}
 		if isKnown(src.PolicySetDefinitionName) {
 			n := src.PolicySetDefinitionName.ValueString()
 			if !az.PolicySetDefinitionExists(n) {
 				return nil, fmt.Errorf("policy set definition %s not found in AlzLib", n)
 			}
 			dst.Properties.PolicyDefinitionID = to.Ptr(fmt.Sprintf(policySetDefinitionIdFmt, n))
+			policyDefinitionMode = "PolicySet"
 		}
 		if isKnown(src.PolicyDefinitionId) {
```
Comment (matt-FFFFFF) [inline internal/provider/archetype_data_source.go:716]: This scenario covers assigning a built-in policy or policy set.

Therefore we need to detect this and handle both cases.

At the moment we are assuming this is not a policy set, though this assumption could be incorrect.

File: internal/provider/archetype_data_source.go [hunk 2]
```diff
@@ -689,23 +689,38 @@ func policyAssignmentType2ArmPolicyAssignment(pamap map[string]PolicyAssignmentT
 		dst.Type = to.Ptr(policyAssignementType)
 		dst.Properties.DisplayName = to.Ptr(src.DisplayName.ValueString())
 
+		policyDefinitionMode := "NotSupported"
```
Comment (matt-FFFFFF) [inline internal/provider/archetype_data_source.go:692]: I wonder if we should use a stronger type than a string here. Thinking of iota or type/const based approach.

**Review by matt-FFFFFF — CHANGES_REQUESTED** [review]
Thanks for this @jaredfholgate. Another good find 🙂 

Couple of comments to consider.

File: internal/provider/archetype_data_source.go (hunk 2 above)
Comment (jaredfholgate) [reply internal/provider/archetype_data_source.go:692]: I am having second thoughts about this change anyway. Can discuss next week. I think the module fix will suffice for now.

**Comment by matt-FFFFFF** [comment]
closing for now, feel free to re-open

**Pull request closed without merging**